packages/fetchai/protocols/fipa/dialogues.py

```diff
@@ -335,18 +327,66 @@ def get_dialogue(self, fipa_msg: Message, agent_addr: Address) -> Dialogue:
             other_initiated_dialogue = cast(
                 FipaDialogue, self.dialogues[other_initiated_dialogue_label]
             )
-            if other_initiated_dialogue.is_valid_next_message(fipa_msg):
-                result = other_initiated_dialogue
+            result = other_initiated_dialogue
         if self_initiated_dialogue_label in self.dialogues:
             self_initiated_dialogue = cast(
                 FipaDialogue, self.dialogues[self_initiated_dialogue_label]
             )
-            if self_initiated_dialogue.is_valid_next_message(fipa_msg):
-                result = self_initiated_dialogue
+            result = self_initiated_dialogue
```

Comment:
Note, we most likely need to perform the is valid next message checks elsewhere then...otherwise some current implementations might fail..

Reply:
The new way, the checks should be done in the extend calls. 
result = incoming/outoing_extend(msg)
if not result:
     # failed
     # error handling
else:
     # succeeded